framework/src/functions/MooseParsedFunctionWrapper.C

```diff
@@ -118,6 +118,12 @@ MooseParsedFunctionWrapper::initialize()
       _function_index.push_back(i);
     }
 
+    // Error when a variable is supplied
+    else if (_feproblem.hasVariable(_vals_input[i]))
+      mooseError("The 'vals' input includes an item ('",
```

Comment:
FYI: mooseParamError doesn't work here because this class is not a MooseObject.

Reply:
Maybe so - but @rwcarlsen is pretty good at figuring out how to make it work. I'll assign this to him.

Reply:
Could we just pass the `MooseObject*` into the MooseParsedFunctionWrapper constructor?  It seems like there are only  a couple of places that would need to be changed.

Reply:
I would vote for changing the constructor - We pass MooseObject in for all of our Interfaces. That's basically what this is.